### ISSUE TITLE
Deprecate accepting None in pd.concat

### ISSUE THREAD

**Issue by twoertwein** [pandas-dev/pandas#57846]
`pd.concat` accepts iterables that may contain `Series`, `DataFrame` and `None`, where `None` are simply ignored.

```py
pd.concat([None, series, None, series, None]) # works, same as pd.concat([series, series])
pd.concat([None]) # raises
```

It would be great to deprecate/remove accepting `None`s. This will help to better annotate `concat` https://github.com/pandas-dev/pandas-stubs/issues/888

Pandas-stubs currently has to choose between either false-positives (accept `concat([None])` ) or false-negatives (reject some `concat(Iterable[Any/Unknown])` that could succeed).

**Comment by phofl** [comment]
I am -1 on simply deprecating this because of typing reasons. Let's think about use cases before we make a decision. We definitely can't do this without a deprecation

**Comment by twoertwein** [comment]
I agree, it would be good first to understand which use case supporting `None` enables!

@Dr-Irv found a way to get it to type check without having to decide between false negatives/positives :)

**Comment by jbrockmendel** [comment]
I was surprised to find this works.  Looks like Nones are filtered out in _clean_keys_and_objs, which has a comment pointing to #1649 back in 2012.  I don't see much conversation there about whether this should be allowed.

+1 on deprecating.
